lib/cocoapods/target/build_settings.rb

```diff
@@ -558,9 +558,8 @@ def initialize(target, non_library_spec = nil)
           if library_xcconfig? && (target.should_build? && target.build_as_dynamic?)
             frameworks.concat vendored_static_frameworks.map { |l| File.basename(l, '.framework') }
           end
-          if non_library_xcconfig?
-            frameworks.concat dependent_targets_to_link.flat_map { |pt| pt.build_settings.frameworks_to_import }
-          end
+          targets_to_link = library_xcconfig? ? dependent_targets.reject(&:should_build?) : dependent_targets_to_link
```

Comment:
for unbbuilt dependent targets, should we only be linking dynamic frameworks here, instead of dynamic and static?

Reply:
Yes, updated.